### PR TITLE
add multi window support

### PR DESCRIPTION
**Pull request by mcc12**

### CODE REVIEW

File: src/components/panels/WindowContainer.tsx [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-import { TabButtonComponent } from './TabButtonComponent';
+import { PanelIconComponent } from './PanelIconComponent';
 import goodboyLogo from '../../assets/goodboy-logo.png';
 import styled from 'styled-components';
 
-export interface MainPanelProps
+export interface WindowContainerData
 {
+    /** container default position */
+    defaultContainerPosition?: { x: string , y: string };
```
Comment (Zyie) [inline src/components/panels/WindowContainer.tsx]: We should probably create a utils file for common types like this. That way we can have an `XY` like in odie

**Review by Zyie — CHANGES_REQUESTED** [review]

File: src/components/panels/WindowContainer.tsx (hunk 1 above)
Comment (mcc12) [reply src/components/panels/WindowContainer.tsx]: thats a good shout

**Review by Zyie — APPROVED** [review]

File: src/components/inputs/layouts/RowLayout.tsx [hunk 2]
```diff
@@ -41,13 +41,13 @@ export const RowLayout: FunctionComponent<RowLayoutProps> = (props: RowLayoutPro
...
+                props.rows.map((row, index) => {
                     const itemWidth = ((100 / row.components.length) + '%');
-                    return <RowContainer>
+                    return <RowContainer key={index}>
                         {
-                            row.components.map((pair) => {
+                            row.components.map((pair, rowIndex) => {
                                 return (
-                                    <ItemContainer width={itemWidth}>
+                                    <ItemContainer key={rowIndex} width={itemWidth}>
```
Comment (maurogb) [inline src/components/inputs/layouts/RowLayout.tsx]: Is it possible to break deeply nested components like this into smaller functions? These functions don't need to be exported, just set in a way to make things easier to read, for instance:
```
function RowContainer(index, pair, rowIndex, itemWidth) {
    return <RowContainer key={index}> {
        row.components.map((pair, rowIndex) => {
            return ItemContainer(pair, rowIndex, itemWidth)
        })
    }
    </RowContainer>
}

function ItemContainer(pair, rowIndex, itemWIdth) {
    return <ItemContainer key={rowIndex} width={itemWidth}>
        <pair.component inputData={pair.inputData}/>
    </ItemContainer>
}
```
Mind that is just matter of a naive opinion, so feel free to disagree or ignore if that doesn't make sense.

Comment (Zyie) [reply src/components/inputs/layouts/RowLayout.tsx]: I think that is a good idea!

File: src/components/panels/WindowContainer.tsx [hunk 3]
```diff
@@ -97,18 +102,19 @@ export const MainPanel: FunctionComponent<MainPanelProps> = (props: MainPanelPro
...
                 {
-                    props.tabData.map((tab, index) => {
+                    props.panelData.map((panel, index) => {
                         return (
-                            <TabButtonComponent
-                                id={tab.id ? tab.id : tab.title.slice(0, 1)}
-                                idColour={tab.idColour}
-                                tabColour={tab.tabColour}
+                            <PanelIconComponent
+                                id={panel.id ? panel.id : panel.title.slice(0, 1)}
```
Comment (maurogb) [inline src/components/panels/WindowContainer.tsx]: Is it safe to use only the first character of title if `panel.id` is not present? Wouldn't be better to convert title into an id-friendly format like `'My Title'` to `'my-title'` and set that as id instead?

**Review by maurogb — APPROVED** [review]

File: src/components/panels/WindowContainer.tsx (hunk 3 above)
Comment (mcc12) [reply src/components/panels/WindowContainer.tsx]: I may need to think of a better name for panel.id, it is being used here as the text inside the panel icon, the full title would be too long for this

File: src/components/inputs/layouts/RowLayout.tsx (hunk 2 above)
Comment (mcc12) [reply src/components/inputs/layouts/RowLayout.tsx]: done

**Pull request merged**